tools/predict_net.py

```diff
@@ -0,0 +1,207 @@
+            frame_processed = imresize(frame_processed, 256)
+            frames.append(frame_processed)
+            
+        if len(frames) == cfg.DATA.NUM_FRAMES*cfg.DATA.SAMPLING_RATE:
+            start = time()
+            # Perform color normalization.
+            inputs = torch.tensor(frames).float()
+            inputs = inputs / 255.0
+            inputs = inputs - torch.tensor(cfg.DATA.MEAN)
+            inputs = inputs / torch.tensor(cfg.DATA.STD)
+            # T H W C -> C T H W.
+            inputs = inputs.permute(3, 0, 1, 2)
+            # 1 C T H W.
+            inputs = inputs[None, :, :, :, :]
+            # Sample frames for the fast pathway.
+            index = torch.linspace(0, inputs.shape[2] - 1, cfg.DATA.NUM_FRAMES).long()
+            fast_pathway = torch.index_select(inputs, 2, index)
+            logger.info('fast_pathway.shape={}'.format(fast_pathway.shape))
+            # Sample frames for the slow pathway.
+            index = torch.linspace(0, fast_pathway.shape[2] - 1, 
+                                fast_pathway.shape[2]//cfg.SLOWFAST.ALPHA).long()
+            slow_pathway = torch.index_select(fast_pathway, 2, index)
+            logger.info('slow_pathway.shape={}'.format(slow_pathway.shape))
+            inputs = [slow_pathway, fast_pathway]
+            # Transfer the data to the current GPU device.
+            if isinstance(inputs, (list,)):
+                for i in range(len(inputs)):
+                    inputs[i] = inputs[i].cuda(non_blocking=True)
+            else:
+                inputs = inputs.cuda(non_blocking=True)
+
+            # Perform the forward pass.
+            preds = model(inputs)
+            # Gather all the predictions across all the devices to perform ensemble.
+            if cfg.NUM_GPUS > 1:
+                preds = du.all_gather(preds)[0]
+
+            logger.info(preds.shape)
+            ## Option 1: single label inference selected from the highest probability entry.
+            label_idx = torch.argmax(preds, -1).cpu()
+            logger.info(label_idx)
+            label = labels[label_idx]
+            ## Option 2: multi-label inferencing selected from probability entries > threshold
+            
+
+            # remove the oldest frame in the buffer to make place for the new one.
+            # frames.pop(0)
+            frames = []
+            s = time() - start
+
+        # Display predicted label to frame
+        cv2.putText(frame, 'Action: {}'.format(label), (20, 30), 
+                    fontFace=cv2.FONT_HERSHEY_SIMPLEX,
+                    fontScale=1, color=(0, 235, 0), thickness=3)
+        # Display prediction speed to frame
+        cv2.putText(frame, 'Speed: {:.2f}s'.format(s), (20, 60), 
+                    fontFace=cv2.FONT_HERSHEY_SIMPLEX,
+                    fontScale=1, color=(0, 235, 0), thickness=3)
+        # Display the frame
+        cv2.imshow('SlowFast', frame)   
```

Comment:
I will move this part to a utility func in demo when we have demo for ava.

Reply:
I made another version to enable multi-label prediction using a threshold, what do you think?